src/macrofier/engine.mjs

```diff
@@ -673,6 +673,24 @@ function generateDefaults(json = {}, templates) {
   return reducer(json)
 }
 
+function sortSchemasByReference(schemas = []) {
+  let indexA = 0;
+  while (indexA < schemas.length) {
+
+    let swapped = false
+    for (let indexB = indexA + 1; indexB < schemas.length; ++indexB) {
+      const bInA = isDefinitionReferencedBySchema('#/components/schemas/' + schemas[indexB][0], schemas[indexA][1])
+      if ((isEnum(schemas[indexB][1]) && !isEnum(schemas[indexA][1])) || (bInA === true))  {
+        [schemas[indexA], schemas[indexB]] = [schemas[indexB], schemas[indexA]]
```

Comment:
Woah, does that work? Impressive trick :)

Reply:
yes, it is sorting properly now :)